### PR TITLE
Add Python Solution to Bucket List

### PR DESCRIPTION
**Pull request by Aadit-Ambadkar**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-856.mdx [hunk 1]
```diff
@@ -208,5 +208,46 @@ public class TheBucketList2 {
 ```
 	
 </JavaSection>
+<PySection>
+```py
+fin = open("blist.in", "r")
+fout = open("blist.out", "w")
+
+event = [0 for i in range(1001)]  # initialize all elements to 0
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:216]:
```suggestion
event = [0 for i in range(1001)]
```
you have the variable `max_time` as well, use it

File: solutions/usaco-856.mdx [hunk 2]
```diff
@@ -208,5 +208,46 @@ public class TheBucketList2 {
...
+	# at time start, cow i needs to be milked, so 
+	# we'll need amt additional buckets (hence adding amt)
+	event[start] += amt
+
+	# at time end, cow i is done milking, so we can free up 
+	# amt buckets (hence subtracting amt)
+	event[end] -= amt
 
+# res represents the maximum number of buckets we'll need 
+# to use at any given time and cur represents
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:233]:
```py
"""
multiline comments should
be like this
"""
```
also at this point there's more comments than code
consider just putting an editorial or something

File: solutions/usaco-856.mdx [hunk 3]
```diff
@@ -208,5 +208,46 @@ public class TheBucketList2 {
...
+	event[end] -= amt
 
+# res represents the maximum number of buckets we'll need 
+# to use at any given time and cur represents 
+# how many buckets we need to use at the current 
+# time we're processing
+res = 0
+cur = 0
+
+for i in range(max_time+1):
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:239]: spaces

File: solutions/usaco-856.mdx [hunk 4]
```diff
@@ -208,5 +208,46 @@ public class TheBucketList2 {
 ```
 	
 </JavaSection>
+<PySection>
+```py
+fin = open("blist.in", "r")
+fout = open("blist.out", "w")
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:214]:
```suggestion
```

File: solutions/usaco-856.mdx [hunk 5]
```diff
@@ -208,5 +208,46 @@ public class TheBucketList2 {
...
+	# so we want to update the number of buckets 
+	# we are currently using accordingly
+	cur += event[i]
+
+	# we want res to be the greatest number of buckets 
+	# we need to use at any given point in time
+	res = max(res, cur)
+
+fout.write(str(res))
+fout.close()
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:250]:
```suggestion
print(res, file=open('blist.out', 'w'))
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-856.mdx (hunk 2 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-856.mdx:233]: lmao check the c++ solution

Comment (SansPapyrus683) [reply solutions/usaco-856.mdx]: then fix the c++ solution as well

File: solutions/usaco-856.mdx [hunk 6]
```diff
@@ -208,5 +208,45 @@ public class TheBucketList2 {
 ```
 	
 </JavaSection>
-
+<PySection>
+```py
+fin = open("blist.in", "r")
+
+max_time = 1001
```
Comment (SansPapyrus683) [inline solutions/usaco-856.mdx:215]: change this to 1000 and use `max_time + 1` where necessary

Comment (Aadit-Ambadkar) [reply solutions/usaco-856.mdx:215]: there is literally no other use of max_time except for max_time + 1

File: solutions/usaco-856.mdx (hunk 2 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-856.mdx]: it's not needed, and the comments fit well anyways

File: solutions/usaco-856.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-856.mdx:215]: still doesn't make the actual value of `max_time` 1001

**Comment by Aadit-Ambadkar** [comment]
@SansPapyrus683 check now?

**Review by SansPapyrus683 — APPROVED** [review]
another approval and we gud

**Review by brad182 — APPROVED** [review]

**Pull request merged**